src/ConfigFileParser.cpp

```diff
@@ -389,28 +440,33 @@ void ConfigFileParser::readSocket(const std::string& value)
 		std::string ipAddress = value.substr(0, colonIndex);
 		if (!webutils::isIpAddressValid(ipAddress))
 			throw std::runtime_error("Invalid ip address");
-		m_configFile.servers[m_serverIndex].host = ipAddress;
+
+		if (ipAddress == "localhost")
+			m_configFile.servers[m_serverIndex].host = "127.0.0.1";
+		else
+			m_configFile.servers[m_serverIndex].host = ipAddress;
 
 		std::string port = value.substr(colonIndex + 1, semicolonIndex - colonIndex - 1);
 		if (!webutils::isPortValid(port))
 			throw std::runtime_error("Invalid port");
 		m_configFile.servers[m_serverIndex].port = port;
 	} else {
 		if (dot == std::string::npos) {
-			std::string port = value.substr(0, semicolonIndex);
-			if (!webutils::isPortValid(port))
-				throw std::runtime_error("Invalid port");
-
-			m_configFile.servers[m_serverIndex].host = "127.0.0.1";
-			m_configFile.servers[m_serverIndex].port = port;
-
+			std::string ipAddress = value.substr(0, semicolonIndex);
+			if (ipAddress == "localhost")
+				m_configFile.servers[m_serverIndex].host = "127.0.0.1";
+			else {
+				std::string port = value.substr(0, semicolonIndex);
+				if (!webutils::isPortValid(port))
+					throw std::runtime_error("Invalid port");
+				m_configFile.servers[m_serverIndex].port = port;
+			}
```

Comment:
Here you are assuming if there is no '.' found in the string that it must be either
1. localhost
2. port
And you are extracting the same substring twice and assigning it to two different strings. I would just do it once in the beginning and give it a more general name, e.g. temp or endpoint.

Optionally could add a comment as well to explain this as it is not evident on first sight.

Reply:
True, could be confusing. 

I use now only the var `hostOrPort` for storing the substring.